LiteCore/Unix/icu_shim.c

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright (C) 2015 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * 
+ * Modified for use on Linux Desktop by Jim Borden <jim.borden@couchbase.com>
+ */
+
+#include "icu_shim.h"
+#include <ctype.h>
+#include <dirent.h>
+#include <dlfcn.h>
+#include <pthread.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+#ifdef CBL_USE_ICU_SHIM
+#define LOCAL_INLINE
+
+// This is not meant to be absolutely foolproof, but only to cover
+// the triples we support on Linux
+#if defined(__x86_64__)
+#define ARCH_FOLDER "x86_64-linux-gnu"
+#elif defined(__aarch64__)
+#define ARCH_FOLDER "aarch64-linux-gnu"
+#elif defined(__ARM_ARCH_7A__)
+#define ARCH_FOLDER "arm-linux-gnueabihf"
+#endif
+
+
```

Comment:
This file looks like a copy of ndk_icu.c, at least most of its code is the same. That's a lot of duplication. Could we do this with a single source file?

Reply:
There is a good chunk of its code that is the same.  I thought it was kind of nice having a straight up copy and paste for ndk_icu.c (which is specific to Android in the way that it looks up the shared library) with the license intact and this modified one for desktop platforms but there could be some refactoring done.  It will need to balance `#ifdef` with separate files as usual though.